README.md

```diff
@@ -30,4 +30,26 @@ so this extension was disabled.  If you care about this feature and are ok with
 Unfortunately, it does not yet support tags (see [issue](https://github.com/Azure/Azurite/issues/647)).  This is problematic because tags is how this library queries for expired files.
 To work around this, I have split out the expiration stuff into a separate interface `ITusExpirationDetailsStore` and implemented `NullExpirationDetailsStore`.
 
-This allows you to use an external database (for just dev, or both prod and dev), or you can use `NullExpirationDetailsStore` in dev to forego using the expiration feature when using Azurite.
\ No newline at end of file
+This allows you to use an external database (for just dev, or both prod and dev), or you can use `NullExpirationDetailsStore` in dev to forego using the expiration feature when using Azurite.
+
+### Version notes
+
+#### Why is the `UploadOffset` property on my upload blob missing or not updating anymore?
+
+In version 3.0.0 the `UploadOffset` property was deprecated in favor of the actual `Content-Length` HTTP property of the blob.
+
+This was done to fix possible upload blob data corruption when resuming uploads after a broken/cancelled connection and has the following (unproblematic) side effects:
+
+* Existing upload blobs created with version < 3.0.0 contain an `UploadOffset` property which is not updated anymore after updating to version ≥ 3.0.0.
+* New upload blobs created with version ≥ 3.0.0 do not contain an `UploadOffset` property at all.
+
+#### Can I prevent blob data corruption with versions < 3.x?
+
+Yes, by choosing a TUS chunk size ≤ 4 MiB (4 194 304 bytes). Please note that such small chunk sizes are not recommended because they reduce upload performance.
```

Comment:
I want to make sure I understand, as I haven't been in the code too much recently. I see this [const](https://github.com/giometrix/Xtensible.TusDotNet.Azure/blob/8d45119247e33f5560463cbb58806e6b2c1ef917/src/Xtensible.TusDotNet.Azure/AzureBlobTusStore.cs#L28) and I don't see a way of overriding (just did a cursory search though).

1. Clearly you had issues, so is it true that the issue only occurs with larger chunks?
2. "Please note that such small chunk sizes are not recommended" , again I may be missing the place where we make chunk sizes configurable.... but I didn't look to closely, just taking a cursory look. If I am correct should we open this up (as not part of this PR)?

To be clear, I do think making the change to use ContentLength is the right thing to do; I just want to make sure that I understand what is being said here and possibility to open up features.

Reply:
When a TUS client is uploading a file it does so in chunks. It could be a single chunk containing the whole file or the file could be split into multiple chunks, which are uploaded separately. Especially when resuming uploads, the chunk size is not the whole file but the remaining file size at most. Each chunk is sent as a HTTP PATCH request from the TUS client to the TUS server.

The chosen TUS chunk size actually depends on the TUS client implementation and configuration, as well as the TUS server limits.

The `ITusStore.AppendDataAsync(...)` method receives a stream of data, which is sent by the TUS client. The length of this stream is the TUS chunk size, that was chosen by the TUS client.

Now, the `AzureBlobTusStore` performs its own splitting of the data, because Azure Append Blobs can only be appended to in blocks. The maximum size of each block is defined by the `AppendBlobBlockSize` as 4 MiB, which were chosen, according to the README, to support checksum validation. (Honestly, I have not yet tried to understand the checksum code yet, so I can't say anything in that regard.)

To answer the actual questions:

1. Actually, the risk of the issue happening is not zero when using a TUS chunk size ≤ 4 MiB. Something (CancellationToken triggered, App killed, Azure connection interrupted) could prevent the UploadOffset Azure Metadata property update from happening after the actual data was appended to the Azure Blob. So the README text should actually read "You cannot fully prevent it, only minimize the risk by ...".
2. This library cannot influence the TUS chunk size in any way. It has to live with the data that arrives at the `AppendDataAsync(...)` method. As explained above, the TUS chunk size is chosen by the TUS client. The README text could be altered to make that clear. I suggest: "... by limiting your TUS client(s) to a chunk size ≤ 4 MiB, in that case it is also suggested to configure tusdotnet to disallow larger chunks.".

Reply:
@giometrix Oh, you already merged this without my suggested changes to the README. Should I submit another pull request?

Reply:
You can if you want, and I’ll merge it.

I didn’t because I don’t think we should make too much of a big deal about upgrading to 3.0; I want to encourage everyone to upgrade, as it’s pretty painless and solves a potential corruption issue many may not have been aware of.

But if you feel like it’s an important note I don’t mind adding it.